libs/api/domains/application/src/lib/application.service.ts

```diff
@@ -103,12 +103,16 @@ export class ApplicationService {
       .catch(handleError)
   }
 
-  async updateExternalData(input: UpdateApplicationExternalDataInput) {
+  async updateExternalData(
+    input: UpdateApplicationExternalDataInput,
+    accessToken?: string,
+  ) {
     const { id, ...populateExternalDataDto } = input
 
     return this.applicationApi.applicationControllerUpdateExternalData({
       id,
       populateExternalDataDto,
+      xAccessToken: accessToken,
```

Comment:
Ideally, we can configure applicationApi with a preMiddleware, that can access `CurrentUser`, and add its access token to the request headers.

Reply:
see answer above